File: numbers/sum_of_numbers/sum_of_numbers.cpp
#include <iostream>

using namespace std;

int sum_of_numbers_iterative(int n){
    int the_sum = 0;

    for(int i=0; i<=n; i++)
        the_sum += i;

    return the_sum;
}

int main(){

	int n;

	cout<<"Enter a number: ";
	cin>>n;

	cout<<sum_of_numbers_iterative(n);  //5+4+3+2+1 = 15

	return 0;
}

File: numbers/sum_of_numbers/sum_of_numbers.go
package main

import "fmt"

func sum_of_numbers_iterative(n int) int {
	the_sum := 0

	for i := 1; i <= n; i++ {
		the_sum += i
	}

	return the_sum
}

func main() {
	var n int
	fmt.Print("Enter a number: ")
	fmt.Scan(&n)

	fmt.Println(sum_of_numbers_iterative(n))
}
